### PR TITLE
Add `Rails/RedundantAllowNil` cop

### PR DESCRIPTION
**Pull request by elebow**
This cop checks for a situation in a Rails model validation where the value of `allow_blank` makes `allow_nil` redundant.

In Rails, `#blank?` includes nil checking. When both options are set to the same value, `allow_nil` is entirely subsumed within `allow_blank`.

When `allow_nil: false, allow_blank: true`, again `allow_blank` controls because Rails performs an `||` between the two. See https://github.com/rails/rails/blob/d5699198a45a91250e1adb3ed899b0b46b4ac879/activemodel/lib/active_model/validator.rb#L151

When `allow_nil: true, allow_blank: false`, both options are meaningful. In this case, a value of `nil` would be valid, but a value of `''` would not be.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ### New features
 
+* Add new cop `Rails/RedundantAllowNil` that checks for cases when `allow_blank` makes `allow_nil` unnecessary in model validations. ([@elebow][])
```
Comment (bbatsov) [inline CHANGELOG.md]: Please move this to the proper section of the changelog. 0.65 was released a while ago.

File: lib/rubocop/cop/rails/redundant_allow_nil.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      #
+      #   # good
+      #   # Here, `nil` is valid but `''` is not
+      #   validates :x, length: { is: 5 }, allow_nil: true, allow_blank: false
+      #
+      class RedundantAllowNil < Cop
+        include RangeHelp
+
+        MSG_SAME =
+          'allow_nil is redundant when allow_blank has the same value'.freeze
```
Comment (bbatsov) [inline lib/rubocop/cop/rails/redundant_allow_nil.rb]: Surround identifiers with backticks and end messages with `.`.

**Comment by bbatsov** [comment]
Apart from my small remarks your changes look good.

File: config/default.yml [hunk 3]
```diff
@@ -2488,6 +2488,13 @@ Rails/ReadWriteAttribute:
   Include:
     - app/models/**/*.rb
 
+Rails/RedundantAllowNil:
+  Description: >-
+                 Finds redundant use of `allow_nil` when `allow_blank` is set to
+                 certain values in model validations.
+  Enabled: true
+  VersionAdded: '0.67'
```
Comment (bbatsov) [inline config/default.yml]: I think you should limit this check to `app/models/**/*.rb`.

**Comment by elebow** [comment]
Updated to address comments, and rebased on latest master.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**